jsx/LoadingBar.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+
+interface LoadingBarProps {
+  progress: number; // Expect progress to be a number
+}
+
+/**
+ * LoadingBar is a React component that displays a progress bar to indicate
+ * loading status.
+ *
+ * @param {number} progress - A number representing the loading progress (0 to
+ * 100).
```

Comment:
Agreed on the redundancy of typing in docs — I didn't type 'prop' because that feels opaque and what feels relevant to be documented are the actual attributes of the props object.

How would you suggest I document it?

Reply:
According to the [JSDoc documentation](https://jsdoc.app/tags-param#parameters-with-properties), you should have one line for `props` and one line for `props.progress`. This is kinda verbose but I guess that's the standard way to do it. But since props is typed with `LoadingBarProps`, I guess just documentating `props` is ok, `progress` is documented in the interface.

Alternatively, if you're patient enough, I'll push for the removal of the `@param` / `@returns` requirement in the JSDoc, I agree it's often redundant with Typescript.